### PR TITLE
SP-4021 use spctl as dependency

### PR DESCRIPTION
**Pull request by Idris0v**

### CODE REVIEW

File: Provider Provisioner/tsconfig.build.json [hunk 1]
```diff
@@ -4,7 +4,9 @@
         "outDir": "./dist",
         "rootDir": "src",
         "composite": true,
-        "tsBuildInfoFile": "./dist/tsconfig.build.tsbuildinfo"
+        "tsBuildInfoFile": "./dist/tsconfig.build.tsbuildinfo",
+        "skipLibCheck": true,
+        "noImplicitAny": false
```
Comment (Idris0v) [inline Provider Provisioner/tsconfig.build.json:9]: Setting noImplicitAny to false helped eliminate errors caused by missing @types/lib libraries.

File: Provider Provisioner/src/spctl/spctl.service.ts
Comment (Idris0v) [inline Provider Provisioner/src/spctl/spctl.service.ts:1]: spctl service is not spawning processes anymore

File: Provider Provisioner/src/handle-input-offer.ts [hunk 2]
```diff
@@ -15,37 +15,34 @@ export async function handleInputOffer(params: HandleInputOfferParams): Promise<
   const { inputOffer, spctlService, logger } = params;
   const orderStatuses = [OrderStatus.New, OrderStatus.Processing];
 
-  for (const orderStatus of orderStatuses) {
```
Comment (Idris0v) [inline Provider Provisioner/src/handle-input-offer.ts:18]: removed this loop

File: Provider Provisioner/src/spctl/spctl.service.ts [hunk 3]
```diff
@@ -1,7 +1,9 @@
 import fs from 'fs/promises';
 import path from 'path';
 import { ILogger } from '../logger';
-import { spawnCommand } from './spawn-command';
+import ordersComplete from 'spctl/build/commands/ordersComplete';
+import ordersList from 'spctl/build/commands/ordersList';
+import ConfigLoader from 'spctl/build/config';
```
Comment (Idris0v) [inline Provider Provisioner/src/spctl/spctl.service.ts:6]: notice here we use build folder from spctl module.

File: Provider Provisioner/scripts/prepare-solution.sh [hunk 4]
```diff
@@ -7,21 +7,14 @@ solution_name=$2
...
 npm ci
 
 echo "Compiling typescript"
-npm run build
+npm run build:all
 
-echo "Remove dev dependendencies"
-npm ci --omit dev
+# echo "Remove dev dependendencies"
+# npm ci --omit dev
```
Comment (marchuk-vlad) [inline Provider Provisioner/scripts/prepare-solution.sh:17]: Do we left dev dependencies intentionally ?

**Review by marchuk-vlad — CHANGES_REQUESTED** [review]
We don't need `download-spctl` scripts and its dependencies like `cli-progress` and `colors` anymore, do we?

**Review by DmitrySmv — APPROVED** [review]

**Review by marchuk-vlad — APPROVED** [review]

**Pull request merged**